src/compiler/sys.ts

```diff
@@ -24,10 +24,403 @@ namespace ts {
 
     export type FileWatcherCallback = (fileName: string, eventKind: FileWatcherEventKind) => void;
     export type DirectoryWatcherCallback = (fileName: string) => void;
+    /*@internal*/
     export interface WatchedFile {
-        fileName: string;
-        callback: FileWatcherCallback;
-        mtime?: Date;
+        readonly fileName: string;
+        readonly callback: FileWatcherCallback;
+        mtime: Date;
+    }
+
+    /* @internal */

```

Comment:
const enum ?

Reply:
Need it not to be const enum so that we can set custom levels. https://github.com/Microsoft/TypeScript/pull/21243/files#diff-cc0eed295cd2bad794329f871b2669fdR90